File: work_1/content/main.py
import random

# 生成5个无符号32位随机整数并写入文件
numbers = [random.randint(0, 2**32-1) for _ in range(5)]
print("生成的随机数:", " ".join(map(str, numbers)))

with open("input.txt", "w") as f:
    f.write(" ".join(map(str, numbers)))

# 从文件读取数据并计算平均值
with open("input.txt", "r") as f:
    data = f.read()
    numbers = list(map(int, data.split()))
    average = sum(numbers) / len(numbers)

# 将平均值写入输出文件
with open("output.txt", "w") as f:
    f.write(str(average))

print("计算的平均值:", average)

File: work_1/content/main.cpp
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <stdint.h>

int main() {
  
    srand(time(NULL));
    
  
    FILE *input_file = fopen("input.txt", "w");
    if (input_file == NULL) {
        perror("无法打开input.txt文件");
        return 1;
    }
    
    uint32_t numbers[5];
    printf("生成的随机数: ");
    for (int i = 0; i < 5; i++) {
        numbers[i] = rand();
        fprintf(input_file, i == 0 ? "%u" : " %u", numbers[i]);
        printf(i == 0 ? "%u" : " %u", numbers[i]);
    }
    printf("\n");
    fclose(input_file);
    
 
    input_file = fopen("input.txt", "r");
    if (input_file == NULL) {
        perror("无法打开input.txt文件");
        return 1;
    }
    
    float sum = 0.0f;
    for (int i = 0; i < 5; i++) {
        fscanf(input_file, "%u", &numbers[i]);
        sum += numbers[i];
    }
    fclose(input_file);
    
    float average = sum / 5.0f;
    
 
    FILE *output_file = fopen("output.txt", "w");
    if (output_file == NULL) {
        perror("无法打开output.txt文件");
        return 1;
    }
    fprintf(output_file, "%f", average);
    fclose(output_file);
    
    printf("计算的平均值: %f\n", average);
    
    return 0;
}

